src/CDbfReader.cpp

```diff
@@ -157,7 +157,7 @@ CDbfReader::ReadDbf(const std::wstring& wstrDbfFilePath)
             std::move(DbtFile),
             Fields,
             DataStart,
-            BlockLength
+            dbtHeader.BlockLength
```

Comment:
This variable looks uninitialized/incorrect when `dbfHeader.Version == DBASE_III_PLUS_VERSION`

I know that I have asked for `_ReadDbt` to return the full `DbtHeader`, and I still support that.
However, when calling that function in `ReadDbf`, you may want to only have a local `uint16_t BlockLength` there, which is initialized to `DBT_DEFAULT_BLOCK_LENGTH` and reset to `dbtHeader.BlockLength` whenever we got a valid `DbtHeader`.

Reply:
Thinking about it, I got a bit confused by the general variable name `BlockLength`.
It should better be called `DbtBlockLength` here and in `CDbfReader` to clearly indicate that it's only relevant for the DBT file.

That said, my suggestion still sounds like the right way to go, even when we don't expect a DBT file for dBASE III DBF files.